go/vt/sqlparser/ast.go

```diff
@@ -1613,8 +1613,19 @@ type PartitionSpecAction int8
 
 // PartitionDefinition describes a very minimal partition definition
 type PartitionDefinition struct {
-	Name       ColIdent
-	ValueRange *PartitionValueRange
+	Name    ColIdent
+	Options *PartitionDefinitionOptions
+}
+
+type PartitionDefinitionOptions struct {
+	ValueRange     *PartitionValueRange
+	Comment        string
+	Engine         *PartitionEngine
+	DataDirectory  string
+	IndexDirectory string
+	MaxRows        int
```

Comment:
@systay @frouioui @harshit-gangal `MaxRows`, `MinRows`, `DataDirectory` etc are fields that are optional in the Partition definition, so just storing the int/string might be insufficient. i.e., not specifying MaxRows is different from MaxRows being 0.
@K-Kumar-01 opted to set the default value to -1 in the parser, but I think that's not a great idea, in the sense that, someone using this struct later, would need to know that -1 means that no value is specified, otherwise the user might think (rightly so) that it is a positive value, leading to errors down the line.

We have two alternatives and wanted to know what is the best of the two that you guys think-
1. Store them as *string/ *int instead, so the default value is nil
2. Introduce a struct for them (one for each, just having a single value) and store a pointer to that struct.

WDYT?

Reply:
That decides it then, @K-Kumar-01, could you make the necessary changes, and then we can merge it